core/src/main/java/io/grpc/internal/ClientTransportFactory.java

```diff
@@ -53,4 +56,68 @@ ConnectionClientTransport newClientTransport(SocketAddress serverAddress, String
    */
   @Override
   void close();
+
+  public static final class ClientTransportOptions {
```

Comment:
I didn't make it immutable, because it's an internal thing and I didn't see much point in it. But I did consider it. I didn't plumb it more deeply into transport constructors because today nothing but netty uses the EagAttributes. And Netty is complicated due to the TransportCreationParamsFilter; I spoke with @carl-mastrangelo and we agreed to "not touch" that interface if possible.

Even if we plumb it into transport constructors, I expect them to copy the pieces they care about and then throw away the object.

Reply:
Fair enough. Please document that ClientTransportOptions is supposed to be consumed and thrown away immediately, not supposed be kept for long or passed downstream.

Reply:
Done. I hope the comments I added are clear.